aws-datastore/src/main/java/com/amplifyframework/datastore/AsyncUtils.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.datastore;
+
+import androidx.annotation.NonNull;
+
+import com.amplifyframework.core.async.Cancelable;
+
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicReference;
+
+import io.reactivex.disposables.Disposable;
+
+/**
+ * Utility class for functions used throughout DataStore.
+ */
+public final class AsyncUtils {
```

Comment:
I've been working hard for us to avoid having any `FooUtils` classes (by name).

Would it make sense to call this class `Cancelables`? (with one `l`, like our `Cancelable`, and unlike the Java concurrency `Cancellable`.)

It would read like:

```
Disposable disposables = Cancelables.asDisposable(cancelable);
```

The advantage here is that `Cancelables` is only ever going to have a few things in it, and they'll all relate to that data type.

`AsyncUtils` could end up being abused in the future, and have all kinds of executors, schedulers, etc., in it., over time.